src/librustc_metadata/diagnostics.rs

```diff
@@ -91,6 +91,204 @@ You need to link your code to the relevant crate in order to be able to use it
 well, and you link to them the same way.
 "##,
 
+E0466: r##"
+Invalid macro import declarations.
+
+Causes of this error:
+
+```ignore
+#[macro_use(a_macro(another_macro))]    // error: invalid import declaration
+extern crate some_crate;
+
+#[macro_use(i_want = "some_macros")]    // error: invalid import declaration
+extern crate another_crate;
+```
+
+This is a syntax error at the level of attribute declarations.
+
+The proper syntax for macro imports is the following:
+
+```ignore
+// // some_crate contains:
+// #[macro_export]
+// macro_rules! get_tacos {
+//     ...
+// }
+//
+// #[macro_export]
+// macro_rules! bring_beer {
+//     ...
+// }
+#[macro_use(get_tacos, bring_beer)]     // imports macros get_tacos and
```

Comment:
Missing empty line before this line (and still too much whitespaces but in here I don't find it problematic).